File: queue/circularQueue.java
//implementing a circular queue 

import java.util.*;

class Circularqueue{
    int[] cqueue;
    int front, rear;
    
    Circularqueue(int size){
        cqueue=new int[size];
        front=-1;
        rear=-1;
    }
    boolean isEmpty(){
        if (front==-1){
            return true;
        }
        else{
            return false;
        }
    }
    boolean isFull(){
        if((front==0 && rear==cqueue.length) || front==rear+1){
            return true;
        }
        else{
            return false;
        }
    }
    void enqueue(int element){
        if (isFull()){
            System.out.println("Queue is full");
        }
        else{
            if(isEmpty()){
                front=0;
            }
            rear=(rear+1)%cqueue.length;
            cqueue[rear]=element;
        }
    }
    void dequeue(){
        if (isEmpty()){
            System.out.println("Queue is empty");
        }
        else{
            System.out.println("Removed : "+cqueue[front]);
            if (front==rear){
                front=-1;
                rear=-1;
            }
            else{
                front=(front+1)%cqueue.length;
            }
        }
    }
    void display(){
        int i;
        for(i=front;i!=rear;i=(i+1)%cqueue.length){
            System.out.print(cqueue[i]+" ");
        }
        System.out.println(cqueue[i]);
    }
    
}
public class Main
{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int s = sc.nextInt();
		Circularqueue cq = new Circularqueue(s);
		cq.enqueue(10);
		cq.enqueue(20);
		System.out.println(cq.isEmpty());
		cq.enqueue(30);
		cq.display();
		cq.dequeue();
		cq.enqueue(40);
		cq.display();
		
	}
}

File: queue/circularQueue.py
class circularQueue:
    def __init__(self,size):
        self.size=size
        self.items=[None for i in range(size)]
        self.front=-1
        self.rear=-1
    
    def isEmpty(self):
        return self.front==-1
    def isFull(self):
        return (self.front==0 and self.rear==self.size) or (self.front==self.rear+1)
    def enqueue(self,element):
        if self.isFull():
            print("Queue is full")
        else:
            if self.isEmpty():
                self.front=0
            self.rear=(self.rear+1)%self.size
            self.items[self.rear]=element
            
    def dequeue(self):
        if self.isEmpty():
            print("Queue is empty")
        else:
            print("Removed ",self.items[self.front])
            if(self.front==self.rear):
                self.front=-1
                self.rear=-1
            else:
                self.front=(self.front+1)%self.size
    
    def display(self):
        if self.isEmpty():
            print("Queue is empty")
        elif self.rear>=self.front:
            print(*self.items[self.front:self.rear+1])
        else:
            print(*self.items[self.front:],end=" ")
            print(*self.items[:self.rear+1])
            
n=int(input())
cq=circularQueue(n)
cq.enqueue(10)
cq.enqueue(20)
cq.enqueue(30)
cq.display()
cq.dequeue()
cq.display()
cq.enqueue(40)
cq.enqueue(50)
cq.display()

